src/main/java/org/gradle/android/VersionNumber.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2012 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.android;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Ordering;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents, parses, and compares version numbers. Supports a couple of different schemes: <ul> <li>MAJOR.MINOR.MICRO-QUALIFIER (the default).</li> <li>MAJOR.MINOR.MICRO.PATCH-QUALIFIER.</li> </ul>
+ *
+ * <p>The {@link #parse} method handles missing parts and allows "." to be used instead of "-", and "_" to be used instead of "." for the patch number.
+ *
+ * <p>This class considers missing parts to be 0, so that "1.0" == "1.0.0" == "1.0.0_0".</p>
+ *
+ * <p>Note that this class considers "1.2.3-something" less than "1.2.3". Qualifiers are compared lexicographically ("1.2.3-alpha" < "1.2.3-beta") and case-insensitive ("1.2.3-alpha" <
+ * "1.2.3.RELEASE").
+ *
+ * <p>To check if a version number is at least "1.2.3", disregarding a potential qualifier like "beta", use {@code version.getBaseVersion().compareTo(VersionNumber.parse("1.2.3")) >= 0}.
+ *
+ * <p>This implementation is copied from Gradle 7.4.2, as this class was originally part of the {@code org.gradle.util} package, but was deprecated.
+ */
+public class VersionNumber implements Comparable<VersionNumber> {
```

Comment:
I would not have made this class public, since it might lead consumers of this plugin to start using this class.

Reply:
I was going to make it package private, but it is used by the `RoomSchemaLocationWorkaround` in the `workarounds` package.

Reply:
I think groovy shouldn't care, but we can move this class's package as needed.

Reply:
There are several more classes that fall into this cateogy. I suggest to leave it as it is for now.